app/auth/decorators.py

```
@@ -86,6 +89,37 @@ def _authenticate_request(user_type: ClerkUserType):
             raise AuthenticationError("Authentication service error", 500)
 
 
+def _authenticate_api_key():
+    """
+    Authenticate API key from request headers
+    Returns: True if authenticated
+    Raises: AuthenticationError if not authenticated
+    """
+    # Get API key from multiple possible headers
+    api_key = request.headers.get('X-API-Key') or request.headers.get('Authorization')
+    
+    if not api_key:
+        raise AuthenticationError("API key required", 401)
+    
+    # Handle Bearer token format
+    if api_key.startswith('Bearer '):
+        api_key = api_key[7:]  # Remove 'Bearer ' prefix
```

Do we want to throw an error otherwise?